src/components/SidebarNav.tsx

```diff
@@ -0,0 +1,239 @@
+import React from "react";
+import classNames from "classnames";
+import styled, { createGlobalStyle } from "styled-components";
+import { LeftArrow } from "./svgs/LeftArrow";
+import { RightArrow } from "./svgs/RightArrow";
+import { breakpoints, colors } from "../theme";
+import { Dialog, Modal } from "react-aria-components";
+import { useMatchMediaQuery } from "../hooks";
+
+const collapsedWidth = "5.6rem";
+const expandedWidth = "24rem";
+
+const GlobalStyle = createGlobalStyle`
+  .react-aria-ModalOverlay {
+    position: fixed;
+    top: 0;
+    left: 0;
+    width: 100vw;
+    height: var(--visual-viewport-height);
+    background: rgba(0 0 0 / .7);
+    z-index: 100;
+  }
+`;
+
+const StyledModal = styled(Modal)`
+  position: fixed;
+  outline: none;
+  left: 0;
+`;
+
+const Nav = styled.nav`
+  --collapsed-width: ${collapsedWidth};
+  --expanded-width: ${expandedWidth};
+  width: var(--expanded-width);
+  display: flex;
+  justify-content: space-between;
+  flex-direction: column;
+  position: relative;
+  box-shadow: -0.2rem 0 0.4rem rgba(0, 0, 0, 0.1) inset;
+  background: ${colors.palette.neutralBright};
+  color: ${colors.palette.neutralThin};
+
+  @media (max-width: 15em) {
+    --expanded-width: 100vw;
+    width: 100vw;
+  }
+
+  &.collapsed {
+    width: 5.6rem;
+  }
+
+  &.mobile {
+    position: fixed;
+    height: 100%;
+  }
+
+  @keyframes expandSidebarNav {
+    from {
+      width: var(--collapsed-width);
+    }
+    to {
+      width: var(--expanded-width);
+    }
+  }
+
+  @keyframes collapseSidebarNav {
```

Comment:
why was animation necessary vs a transition?

Reply:
For mobile, I needed to remount it under a Modal, so the transition doesn't get a chance to play. This fakes the transition, running the expand on mount inside the modal and the collapse on normal mount. Thankfully it still worked too for the desktop case, though I'm realizing now I probably could have kept the transition for that case and kept the keyframe stuff only for mobile...

Reply:
ah of course, that makes sense